src/features/pdf/PDFView.tsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+
+import cn from 'classnames';
+
+import { useAppSelector } from 'src/common/hooks';
+import AutomaticPDFLayout from 'src/features/pdf/AutomaticPDFLayout';
+import CustomPDFLayout from 'src/features/pdf/CustomPDFLayout';
+import css from 'src/features/pdf/PDFView.module.css';
+import { ReadyForPrint } from 'src/shared/components/ReadyForPrint';
+import { getInstanceContextSelector } from 'src/utils/instanceContext';
+import type { IInstanceContext } from 'src/types/shared';
+
+interface PDFViewProps {
+  appName: string;
+  appOwner?: string;
+}
+
+const PDFView = ({ appName, appOwner }: PDFViewProps) => {
+  const layouts = useAppSelector((state) => state.formLayout.layouts);
+  const excludePageFromPdf = useAppSelector((state) => new Set(state.formLayout.uiConfig.excludePageFromPdf));
+  const excludeComponentFromPdf = useAppSelector((state) => new Set(state.formLayout.uiConfig.excludeComponentFromPdf));
+  const pageOrder = useAppSelector((state) => state.formLayout.uiConfig.tracks.order);
+  const hiddenPages = useAppSelector((state) => new Set(state.formLayout.uiConfig.tracks.hidden));
+  const pdfLayoutName = useAppSelector((state) => state.formLayout.uiConfig.pdfLayoutName);
+  const optionsLoading = useAppSelector((state) => state.optionState.loading);
+  const dataListLoading = useAppSelector((state) => state.dataListState.loading);
+  const repeatingGroups = useAppSelector((state) => state.formLayout.uiConfig.repeatingGroups);
+  const instanceContextSelector = getInstanceContextSelector();
+  const instanceContext: IInstanceContext = useAppSelector(instanceContextSelector);
+  const applicationSettings = useAppSelector((state) => state.applicationSettings.applicationSettings);
+  const formData = useAppSelector((state) => state.formData?.formData);
+  const hiddenFields = useAppSelector((state) => state.formLayout.uiConfig.hiddenFields);
+  const parties = useAppSelector((state) => state.party.parties);
+  const language = useAppSelector((state) => state.language.language);
+  const textResources = useAppSelector((state) => state.textResources.resources);
+  const instance = useAppSelector((state) => state.instanceData.instance);
+  const allOrgs = useAppSelector((state) => state.organisationMetaData.allOrgs);
+  const profile = useAppSelector((state) => state.profile.profile);
+
+  if (
+    optionsLoading ||
+    dataListLoading ||
+    !layouts ||
+    !excludePageFromPdf ||
+    !excludeComponentFromPdf ||
+    !pageOrder ||
+    !hiddenPages ||
+    !repeatingGroups ||
+    !instanceContext ||
+    !applicationSettings ||
+    !formData ||
+    !hiddenFields ||
+    !parties ||
+    !language ||
+    !textResources ||
+    !instance ||
+    !allOrgs ||
+    !profile
+  ) {
+    return null;
+  }
+
+  const pdfLayout = pdfLayoutName ? layouts[pdfLayoutName] : undefined;
+
+  document.body.style.backgroundColor = 'white';
```

Comment:
Not very react friendly/reusable..? Won't this affect future re-renders when this component is not rendered? This should probably be wrapped in a `React.useEffect` that resets the state upon unmounting. If `PrintButton` renders this component in the background (but sets the global CSS to just print this content and nothing else), it will be very confusing why the page bakground changes to white as soon as you press the print button (and never changes back).

Reply:
oh right, forgot about that one 😅 , for some reason it did not affect future renders when going back and fourth between PDF and regular view. So I totally forgot it like that

Reply:
So apparently the reason it works is that when not in the PDF-view, it uses the `Presentation` component, which does the same thing 😅 
![image](https://user-images.githubusercontent.com/47412359/210336157-dd5558e9-7dbe-4d05-9ef4-78bdd721f3f1.png)

Reply:
Gosh, never mind then! :facepalm: